src/TLuaInterpreter.cpp

```diff
@@ -11991,66 +12027,101 @@ int TLuaInterpreter::setFgColor(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#setBgColor
 int TLuaInterpreter::setBgColor(lua_State* L)
 {
-    int s = 0;
-    int n = lua_gettop(L);
-    QString windowName = QStringLiteral("main");
-    int luaRed;
-    int luaGreen;
-    int luaBlue;
-    if (n > 3) {
-        if (!lua_isstring(L, ++s)) {
-            lua_pushfstring(L, "setBgColor: bad argument #%d type (window name as string expected, got %s!)", s, luaL_typename(L, s));
+    Host* pHost = &getHostFromLua(L);
+    QString windowName;
+    int r, g, b, alpha;
+
+    auto validRange = [](int number) { return number >= 0 and number <= 255; };
```

Comment:
Maybe it's more readable with the and [synonym](https://en.wikipedia.org/wiki/Operators_in_C_and_C%2B%2B#C.2B.2B_operator_synonyms), no strong opinion, but we've used `&&` everywhere else :)